### PR TITLE
Remove redundant (null) initializations

### PR DESCRIPTION
**Pull request by mawiesne**
Change
-
- removes redundant (null) initializations
- removes redundant int initializations
- removes redundant close() methods from try-with-resources handled code blocks
- clears some unused imports

### CODE REVIEW

**Comment by mawiesne** [comment]
FYI: @garydgregory @kinow

**Review by kinow — APPROVED** [review]

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected void setNestedEntries(final ByteCode byteCode, final OperandManager op
         // the spec except for multianewarray, which has
         // its own form.)
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java:65]: @mawiesne 
Why is this a standalone declaration and not assigned? IOW, why not perform the assignment from the line below here? Same for the other files changed.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @mawiesne 
Thank you for your PR.
See my comment.

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java (hunk 1 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java:65]: > @mawiesne 
> 
> Why is this a standalone declaration and not assigned? IOW, why not perform the assignment from the line below here? Same for the other files changed.

Will check/optimize that.

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassSpecificReferenceForm.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public ClassSpecificReferenceForm(final int opcode, final String name, final int
     protected void setNestedEntries(final ByteCode byteCode, final OperandManager operandManager, final int offset)
         throws Pack200Exception {
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassSpecificReferenceForm.java:48]: @mawiesne  How about here?

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/InitMethodReferenceForm.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ protected int getPoolID() {
     protected void setNestedEntries(final ByteCode byteCode, final OperandManager operandManager, final int offset)
         throws Pack200Exception {
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/InitMethodReferenceForm.java:52]: @mawiesne  How about here?

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/NewInitMethodRefForm.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ protected String context(final OperandManager operandManager) {
     protected void setNestedEntries(final ByteCode byteCode, final OperandManager operandManager, final int offset)
         throws Pack200Exception {
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/NewInitMethodRefForm.java:44]: @mawiesne  How about here?

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ReferenceForm.java [hunk 5]
```diff
@@ -60,7 +60,7 @@ public void setByteCodeOperands(final ByteCode byteCode, final OperandManager op
     protected void setNestedEntries(final ByteCode byteCode, final OperandManager operandManager, final int offset)
         throws Pack200Exception {
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ReferenceForm.java:63]: @mawiesne  How about here?

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/StringRefForm.java [hunk 6]
```diff
@@ -52,8 +51,8 @@ protected int getPoolID() {
     protected void setNestedEntries(final ByteCode byteCode, final OperandManager operandManager, final int offset)
         throws Pack200Exception {
         final SegmentConstantPool globalPool = operandManager.globalConstantPool();
-        ClassFileEntry[] nested = null;
-        nested = new ClassFileEntry[] {((CPString) globalPool.getValue(getPoolID(), offset))};
+        ClassFileEntry[] nested;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/StringRefForm.java:54]: @mawiesne  How about here?

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @mawiesne Please my comments.

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/StringRefForm.java (hunk 6 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/StringRefForm.java]: Fixed all of them.

File: src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java (hunk 1 above)
Comment (mawiesne) [reply src/main/java/org/apache/commons/compress/harmony/unpack200/bytecode/forms/ClassRefForm.java]: All fixed with latest push.

**Pull request merged**